### ISSUE TITLE
Invite all members of a space

### ISSUE THREAD

**Issue by olekristensen** [humhub/meeting-issues#25]
We are really happy with the meeting module in my housing association, but it becomes tiresome and error prone to invite the same group members to a meeting by typing in all participant's names for eg. reccuring board meetings.

I would suggest searching the groups when typing, so that 'Board' could come up, and then when pressing enter it could expand to the group members, or let the group itself become a token in the participants box.

Alternatively just having the option to invite all members of a space to a meeting would make a lot of sense.

**Comment by buddh4** [comment]
Can't you currently clone meetings, not sure if this clones the participants though.

**Comment by olekristensen** [comment]
Yes, we do clone meetings, but the participants are not cloned, and i would prefer a way to promote user groups and/or page members as addressable.

In general it is a limitation that groups of users are not addressable throughout humhub, also for messages where it would be useful for directing messages to the 'Board' or a project group or subcommittee etc. I realise that pages and groups are not the same, and that a page could have posts open for non-members (public), but there are lots of cases where opening a page for public posting won't work for messages to a group with privacy concerns, eg. complaints etc.

**Comment by buddh4** [comment]
Not sure if you talk about user groups or spaces, in case you are talking about user groups:

Yes user groups are currently not involved in social aspects of the network and only used for permission management and other administrative purposes e.g. setting default spaces. I think it could get confusing to add user-group support to such features and also increases the complexity of the software. Furthermore depending on the use-case one user-group may contain almost all users of a network and addressing thousands of users at once can get problematic and may also be potentially abused.

If you talk about spaces:

Some modules support addressing all users of a space, e.g. adding all members to a calendar event for example. I see adding a similar feature to the meeting or other modules. But I don't think adding space choosers for addressing whole spaces is the way to go at least until there is an eloberate concept in the core. Usually the users you want to address within content of a space (e.g. meeting), are members of this space. For addressing a selection of spaces maybe a module would be the right tool, something like https://www.humhub.com/de/marketplace/mass_notification/

**Comment by olekristensen** [comment]
I see - 

One way around it might be a solution to introduce a core member-picker component for these cases, where typing the first three of a name are to demanding, lots of times you simply don't remember the names of people, but want to invite certain groups.

I gather that letting the model only represent members is desirable. So why not solve it by adding a popup with a member-picker, or an extended dropdown from the member field, that lets you see the entire catalogue of groups and spaces, in a collapsible tree structure, and pick from that. It could be filterable, or even complete. Like a [...], [+] or library icon, with a structured member picker, that would return the relevant list of selected members to the field. That way search strategies are separated from the modules' internals that can continue to stick with flat lists of members. And users can find each other based on tags/pages/groups or other arbitrary properties and affiliations.

best / o

**Comment by buddh4** [comment]
Feel free to open a feature request ideally with a concept in the core repository and describe how such a picker could look like. Generally its planned to improve such pickers by providing suggested results prior of typing a name. This initial selection will depend on the picker context, e.g. a task module could suggest users which where previously assigned to a task etc. But your suggestion seems to go beyond a simple picker widget.

**Comment by olekristensen** [comment]
Anyway in short term an 'Select all members of this space' button on level with the 'Select me' would be fine. It could even be disablable from admin.

This would allow users to invite all in a space, and then maby delete some - but then they don't unintentionally miss some in their meeting invites...

/ o

**Comment by olekristensen** [comment]
yes, i'll try to find time to sketch something ...

/ o

**Comment by samuk** [comment]
@olekristensen did you take this conversation elsewhere? I'm also interested in this functionality.
